master/pkg/searcher/custom_searcher_events_queue.go

```diff
@@ -17,24 +18,69 @@ type (
 	SearcherEventQueue struct {
 		events     []*experimentv1.SearcherEvent
 		eventCount int32 // stores the number of events in the queue.
+		watchers   map[uuid.UUID]chan<- []*experimentv1.SearcherEvent
 	}
 
 	searcherEventQueueJSON struct {
 		EventsJSON []json.RawMessage `json:"custom_searcher_events"`
 		EventCount int32             `json:"custom_searcher_event_count"`
 	}
+
+	// EventsWatcher has a channel which allows communication to the GET searcher events API.
+	EventsWatcher struct {
+		ID uuid.UUID
+		C  <-chan []*experimentv1.SearcherEvent
+	}
 )
 
 func newSearcherEventQueue() *SearcherEventQueue {
 	events := make([]*experimentv1.SearcherEvent, 0)
-	return &SearcherEventQueue{events: events, eventCount: 0}
+	return &SearcherEventQueue{
+		events: events, eventCount: 0,
```

Comment:
I think I'll have to remove all of them or specify all of them. I think there's a benefit to specifying them because it makes it clear. It's consistent with other new search methods such as asha etc.